activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -727,29 +739,11 @@ def configure_connection
 
           super
 
-          # Enforce foreign key constraints
-          # https://www.sqlite.org/pragma.html#pragma_foreign_keys
-          # https://www.sqlite.org/foreignkeys.html
           raw_execute("PRAGMA foreign_keys = ON", "SCHEMA")
-          unless @memory_database
-            # Journal mode WAL allows for greater concurrency (many readers + one writer)
-            # https://www.sqlite.org/pragma.html#pragma_journal_mode
-            raw_execute("PRAGMA journal_mode = WAL", "SCHEMA")
-            # Set more relaxed level of database durability
-            # 2 = "FULL" (sync on every write), 1 = "NORMAL" (sync every 1000 written pages) and 0 = "NONE"
-            # https://www.sqlite.org/pragma.html#pragma_synchronous
-            raw_execute("PRAGMA synchronous = NORMAL", "SCHEMA")
-            # Set the global memory map so all processes can share some data
-            # https://www.sqlite.org/pragma.html#pragma_mmap_size
-            # https://www.sqlite.org/mmap.html
-            raw_execute("PRAGMA mmap_size = #{128.megabytes}", "SCHEMA")
+
+          (@config.keys & PRAGMA_STATEMENTS).each do |key|
```

Comment:
Why limit the PRAGMAs that can be passed from the `database.yml` file? If we are already moving the SQLite configuration into the file, we should grant developers the power to configure other SQLite PRAGMA statements as well.

Reply:
We shouldn't limit them, but the thought behind this approach is to only allow the pragma's supported by the adapter and avoid arbitrarily executing database statements.